manager/allocator/allocator.go

```diff
@@ -32,6 +32,9 @@ type Allocator struct {
 
 	// pluginGetter provides access to docker's plugin inventory.
 	pluginGetter plugingetter.PluginGetter
+
+	// DefaultAddrPool specifies default subnet pool for global scope networks
+	DefaultAddrPool string
```

Comment:
As commented in 
https://github.com/docker/cli/pull/1233#discussion_r205287615, I'm really not a fan of passing the literal string that's passed on the CLI all the way from CLI to moby API to swarmkit and parsing it here.

If we need a list of address pools here, we should pass a list of address-pools over the API

Reply:
We had gone through the CLI format back and forth on our design document and came up with this format. I should have invited you for design document review.   we do have similar format 
` --advertise-addr string           Advertised address (format: <ip|interface>[:port])`

Do you anticipate any issue down the like in this format ? We could talk offline and discuss in details.  Pls let me know.

Reply:
The `--advertise-addr` originally only took `<ip>:<port>`, but such micro-formats caused us a lot of headaches in the past (`-p [ip][port or range of ports or nothing][port or range of ports]`), which is the reason `--publish` now also has a long form syntax.

Format for the cli / flag is something to be discussed in the docker/cli repository, and separate from how the API sends this information. For example, `-p 100-101:80-81` on the command-line will send that information as;

```json
            "Ports": {
                "80/tcp": [
                    {
                        "HostIp": "0.0.0.0",
                        "HostPort": "100"
                    }
                ],
                "81/tcp": [
                    {
                        "HostIp": "0.0.0.0",
                        "HostPort": "101"
                    }
                ]
            },
```

on the API


the CLI format doesn't allow for any expansion (if needed), but also puts syntax validation very deep in the stack

Reply:
I can move  syntax validation from swarmkit to moby code if thats matters. By looking at existing code, all IP address string validation is done in swarmkit code.  I followed the similar path to be consistent. Here we have simple list of IP addresses and optional subnet size(user can omit if they prefer default size 24). So sending as string from CLI, I still feel it doesn't harm or break anything.

Reply:
We are doing validation in moby instead of swarmkit like the way we validate other swarm init params.